django-rgd/rgd/utility.py

```diff
@@ -45,6 +45,20 @@ def get_cache_dir():
     return path
 
 
+def get_lock_dir():
+    path = Path(get_temp_dir(), 'file_locks')
+    path.mkdir(parents=True, exist_ok=True)
+    return path
+
+
+def get_file_lock(path):
+    """Create a file lock under the lock directory."""
+    hash = hashlib.sha512(str(path).encode('utf-8')).hexdigest()
```

Comment:
sha512 is probably overkill

Reply:
Would a standard hash of the string suffice, e.g. `hash(path)`? If it's good enough for a `dict` it should be good enough for us

Reply:
Actually, it looks like Pathlib has hashing implemented:

https://github.com/python/cpython/blob/5b7c7cb104163a178e9d70cb3c80cbfa6af8fbfc/Lib/pathlib.py#L666